pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -103,7 +107,7 @@ func (pl *CSILimits) Filter(ctx context.Context, _ *framework.CycleState, pod *v
 
 	attachedVolumes := make(map[string]string)
 	for _, existingPod := range nodeInfo.Pods {
-		if err := pl.filterAttachableVolumes(csiNode, existingPod.Pod.Spec.Volumes, existingPod.Pod.Namespace, attachedVolumes); err != nil {
+		if err := pl.filterAttachableVolumes(existingPod.Pod, csiNode, false /* existing pod */, attachedVolumes); err != nil {
```

Comment:
@msau42 looking at this filter, we recompute the volume count on the node for each incoming pod, we should cache that in the scheduler's nodeInfo like we do for other resources (memory, cpu etc.).

also, it seems some the checks below are agnostic to the node (like checking the incoming pod PVC and computing its volume count); those can be factored out into a PreFilter instead of redundantly doing them for each node.

Reply:
> we should cache that in the scheduler's nodeInfo like we do for other resources (memory, cpu etc.).

+1 to have a struct caching the storage related attributes to reduce duplicated computation. It can be implemented in a follow-up, and better first aggregate the requirements from storage.

Reply:
created https://github.com/kubernetes/kubernetes/issues/101803